### DISCUSSION TITLE
How to use user defined functions on column conversion  in streamload?

### DISCUSSION THREAD

**Discussion by swiftlei** [apache/doris#41257] [A - General / Q&A]
DORIS VERSION: 2.1.6

As it was told in the office doc that  the column conversion supports using user defined functions in the expression during streamload,  but when I tried to put data in streamload mode, doris would return an error message like "...Does not support non-builtin functions, or function does not exist..."

Running environment:
1、through streamload http-put and the user is root；
2、the streamload data format was json (with read_json_by_line = true)
3、the http header name: columns;  http header content：field1, field2, field3 ..., field4 = myudf1(field1), field5 = myudf2(field1)

When I used some builtin functions such as lower instead of myudf1 or myudf2, it would work correctly.

**Comment by ixzc** [comment]
can you provide your error log?

**Reply by swiftlei** [reply]
This is the related log line, while f_card_id_to_dob is a user defined function which accepts a string parameter and returns string, and the content of http header "columns" is like "field1, field2, ..., field4=f_card_id_to_dob(field1)":

W20240924 12:39:55.208279 1599577 stream_load.cpp:652] plan streaming load failed. errmsg=[ANALYSIS_ERROR]TStatus: errCode = 2, detailMessage = Does not support non-builtin functions, or function does not exist: f_card_id_to_dob(<slot 30>)id=804ce0090c33e641-2cfc236ae05a5eaf, job_id=-1, txn_id=37148, label=b7316566-e7ae-4e9e-a33a-55362f28ac7f, elapse(s)=0

**Reply by swiftlei** [reply]
PS. This user defined function works well in sql such as select f_card_id_to_dob(field1) and so on
